### PR TITLE
controller: update storageclient connection fail log

### PR DESCRIPTION
**Pull request by rohan47**
updated storageclient connection fail log to show provider endpoint as well

### CODE REVIEW

File: internal/controller/storageclient_controller.go [hunk 1]
```diff
@@ -367,7 +367,7 @@ func (r *StorageClientReconciler) newExternalClusterClient() (*providerClient.OC
 	ocsProviderClient, err := providerClient.NewProviderClient(
 		r.ctx, r.storageClient.Spec.StorageProviderEndpoint, time.Second*10)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create a new provider client: %v", err)
+		return nil, fmt.Errorf("failed to create a new provider client while connecting to %v: %v", r.storageClient.Spec.StorageProviderEndpoint, err)
```
Comment (leelavg) [inline internal/controller/storageclient_controller.go:370]: we need this at all places where we are creating new client, like, in claim controller and in heartbeat as well.

Comment (rohan47) [reply internal/controller/storageclient_controller.go:370]: Updated

Comment (leelavg) [reply internal/controller/storageclient_controller.go:370]: Did you check other places as well? I mentioned what I remembered.

Comment (rohan47) [reply internal/controller/storageclient_controller.go:370]: I checked where `NewProviderClient` is created

**Review by leelavg — APPROVED** [review]

**Pull request merged**